data/transactions/asset.go

```diff
@@ -305,16 +305,26 @@ func (ct AssetTransferTxnFields) apply(header Header, balances Balances, spec Sp
 		}
 	}
 
+	// Are we spending our whole balance to the asset creator?
+	var closeToCreator bool
+	if (ct.AssetReceiver == ct.XferAsset.Creator) && (ct.AssetCloseTo == ct.XferAsset.Creator) {
```

Comment:
If you put this inside the `ct.AssetCloseTo != (basics.Address{}) {` case below, you won't need to worry about `AssetReceiver == Creator`, which seems simpler and closer to what I think we'd want to enforce.  Why allow a separate `AssetReceiver` transfer along with the close?

Reply:
Oh I see, and `takeOut` should still succeed even though we're frozen because `ct.AssetAmount` better be zero if we're frozen. Cool

Reply:
Changed, also added a comment mentioning that zero transfers succeed on frozen accounts.